### PR TITLE
Create the filing for SP/GP/12092

### PR DESCRIPTION
**Pull request by meganawong**
*Issue #:* /bcgov/entity#12092

*Description of changes:*
* add 'Put Back On' option for historical companies and 'Administrative Dissolution' option for active companies to 'Add Staff Filing' drop-down
* implement modal for 'Put Back On' and 'Administrative Dissolution' 

**PR for Review - NOT ready for merge**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/Dashboard/StaffNotation.vue
Comment (severinbeauvais) [inline src/components/Dashboard/StaffNotation.vue:90]: Should this also be disabled when prop `disabled` is True?

Comment (severinbeauvais) [reply src/components/Dashboard/StaffNotation.vue:90]: Ok I get it now.

I recommend using a v-template for `isFirm`, and in that block put these 2 v-ifs (isHistorial and !isHistorical). I think that might more clearly bundle this functionality and make it easier to understand. What do you think?

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [inline src/components/dialogs/AddStaffNotationDialog.vue:8]: Very nice and clean and clear 👍

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [inline src/components/dialogs/AddStaffNotationDialog.vue:16]: For clarity, please put the compound conditional in parentheses, and add a space before the ? .

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [inline src/components/dialogs/AddStaffNotationDialog.vue:137]: Please add both of these to state-getters.ts and import them here instead.

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [inline src/components/dialogs/AddStaffNotationDialog.vue:262]: Please use built-in Vuetify class in the template instead of this.

https://vuetifyjs.com/en/styles/text-and-typography/

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [inline src/components/dialogs/AddStaffNotationDialog.vue:259]: Most of our HTML uses top whitespace (on the next component) instead of bottom whitespace.

It's better to stick with all-top or all-bottom whitespace for layout purposes to handle conditional blocks (and also, as a de facto style, easier to update in the future).

**Review by severinbeauvais — COMMENTED** [review]
Please remember to update the app version.

Please add some unit tests. In particular:
- you have new display scenarios in AddStaffNotationDialog depending on props and state
- you have added some filing types (see enum-mixin.spec.ts)

File: src/enums/filingTypes.ts
Comment (severinbeauvais) [inline src/enums/filingTypes.ts:22]: Are there any mixins methods to update? `isTypeXXX' and `filingTypeToName`

File: src/components/Dashboard/StaffNotation.vue
Comment (meganawong) [reply src/components/Dashboard/StaffNotation.vue]: yes that makes sense, the change has been made

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (meganawong) [reply src/components/dialogs/AddStaffNotationDialog.vue]: fixed

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (meganawong) [reply src/components/dialogs/AddStaffNotationDialog.vue:138]: 'name' is passed as a prop from the parent so I don't think the functions can be moved to state-getters.ts and imported

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (meganawong) [reply src/components/dialogs/AddStaffNotationDialog.vue]: done

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (meganawong) [reply src/components/dialogs/AddStaffNotationDialog.vue]: switched to 'padding-top' for component below (notation-text)

File: src/enums/filingTypes.ts
Comment (meganawong) [reply src/enums/filingTypes.ts:22]: added mixins methods for new Filing Types

File: src/components/Dashboard/StaffNotation.vue
Comment (severinbeauvais) [inline src/components/Dashboard/StaffNotation.vue:90]: Please follow existing code style.
1. use " for v-if statements (in template)
2. indent nested blocks

**Comment by severinbeauvais** [comment]
> **PR for Review - NOT ready for merge**

In this case, click the text button in the Reviewers panel (top right) to convert this PR to a draft.

When this PR is ready for review/merge, then click the button down here somewhere to mark it "ready for review".

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [inline src/components/dialogs/AddStaffNotationDialog.vue:251]: Do you need the padding as well as the margin? (The margin uses class helper "mt-2" in the template.) You may be able to use just one of these.

PS We try to use the class helpers as much as possible. In this case the closest one would be "pt-4" (16px). Is that close enough or does UI/UX want exactly 15px?

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [inline src/components/dialogs/AddStaffNotationDialog.vue:25]: You have this same logic on line 17. Would it be possible to remove from both places and add a local getter instead?

File: src/enums/filingNames.ts
Comment (severinbeauvais) [inline src/enums/filingNames.ts:23]: Please sort by key.

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [inline src/components/dialogs/AddStaffNotationDialog.vue:133]: Since this component already imports mixins, you can import the Enum Mixin and use the new filing type helpers you added, instead of duplicating them here 😃 

(The helpers expect an object with property 'name', so you could call it like this: `isTypeAdministrativeDissolution({ name: this.name })`.)

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [inline src/components/dialogs/AddStaffNotationDialog.vue:5]: Oh no! My fault. This became more complicated than I thought.

Please continue to have the local getter (so that this template code is clean), and in the getter just call the global state getter.

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [reply src/components/dialogs/AddStaffNotationDialog.vue]: Nice fix!

File: src/enums/filingNames.ts
Comment (severinbeauvais) [reply src/enums/filingNames.ts:23]: Thanks.

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [reply src/components/dialogs/AddStaffNotationDialog.vue:26]: I was thinking it would be a getter for the string itself. `notationLabel` maybe?

(Lines 17 and 25 aren't consistent in what is displayed in the else case, so I leave it to you to work that out.)

In case you're stuck and there is higher-priority work to do, I think this isn't serious to block this PR. If you decide not to do this now then just add a comment to that effect here.

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [reply src/components/dialogs/AddStaffNotationDialog.vue:5]: Or... using a JS object property shortcut, you could use this in the template instead:

`v-if="istypeAdministrativeDissolution({ name })"`

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [reply src/components/dialogs/AddStaffNotationDialog.vue:26]: Looks good.

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [reply src/components/dialogs/AddStaffNotationDialog.vue]: Looks good.

**Comment by meganawong** [comment]
When I tried using the global state getter in the local getter as (this.isTypeAdministrativeDissolution({ name: this.name }) and (this.isTypePutBackOn({ name: this.name }), both conditions always appeared to evaluate to true and would render both cases? If I pass **name** in as a parameter to both local getters, it behaves as expected but I am not sure if this is a desired implementation. Please let me know your thoughts

**Comment by severinbeauvais** [comment]
> both conditions always appeared to evaluate to true and would render both cases? If I pass **name** in as a parameter to both local getters, it behaves as expected but I am not sure if this is a desired implementation. Please let me know your thoughts

The global getters are like variables, not like functions. Try without `()`.

**Comment by severinbeauvais** [comment]
> > both conditions always appeared to evaluate to true and would render both cases? If I pass **name** in as a parameter to both local getters, it behaves as expected but I am not sure if this is a desired implementation. Please let me know your thoughts
> 
> The global getters are like variables, not like functions. Try without `()`.

This isn't going to work anyway, since the global getters look at the CURRENT filing name, and here you want to determine if it's a given name. Looks like your original local getter solution was the correct one. (Sorry)

**Comment by severinbeauvais** [comment]
> ~This isn't going to work anyway, since the global getters look at the CURRENT filing name, and here you want to determine if it's a given name. Looks like your original local getter solution was the correct one. (Sorry)~

I am quickly making mistakes and slowly remembering where you're at.

Those global calls are indeed FUNCTIONS. They come from a mixin. Ignore my comments above 🙄 

Now I will look at your code again and try to see why it doesn't work.

**Comment by severinbeauvais** [comment]
Try changing administrativeDissolution() to a local getter. Then you can also remove the parameters in the template.

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [inline src/components/dialogs/AddStaffNotationDialog.vue:6]: Move this to previous line.

Comment (meganawong) [reply src/components/dialogs/AddStaffNotationDialog.vue]: done.

File: src/components/Dashboard/StaffNotation.vue
Comment (severinbeauvais) [inline src/components/Dashboard/StaffNotation.vue:97]: Move this to previous line (or if necessary, indent it as per existing code style).

**Review by shabeeb-aot — APPROVED** [review]

**Pull request merged**
